components/VideoTable.vue

```diff
@@ -1,61 +1,121 @@
-          <v-btn small :to="`/watch/${item.id}`">Watch</v-btn>
-          <v-btn small :to="`/admin/videos/${item.id}/edit`">Edit</v-btn>
-          <v-btn small @click="deleteVideo(item)">Delete</v-btn>
-        </td>
-      </template>
-      <template #expanded-item="{headers,item}">
-        <td :colspan="headers.length">
-          <v-row>
-            <v-col cols="12" md="4">
-              <VideoWatch :video="item" />
-            </v-col>
-            <v-col cols="12" md="8">
-              <h1>{{item.name}}</h1>
-              <MarkdownDisplay :markdown="item.description" />
-            </v-col>
-          </v-row>
-        </td>
-      </template>
-    </v-data-table>
+        <v-data-table :items="itemsFilteredByTitleOrDescription"
+                      :headers="headers"
+                      :show-expand="showExpand"
+                      sort-by="sortable_published_at"
+                      :dense="dense"
+                      :items-per-page="itemsPerPage || 10"
+                      @click:row="goToVideo"
+                      :mobile-breakpoint="0"
+                      :page="Number($route.query.pageNumber) || 1"
+                      @update:page="changePageNumber"
+                      :sort-desc="sortDesc"
+                      height="500px"
+        >
+          <template #item.duration="{value, item}">
+            <DurationDisplay :duration="value" />
+          </template>
+          <template #item.sortable_published_at="{item}">
+            <DateDisplay :date="item.published_at" />
+          </template>
+          <template #item.played="{item}">
+            <div class="green--text" v-if="isPlayed(item.id)">
+              <font-awesome-icon icon="check" />
+            </div>
+          </template>
+          <template #item.pro="{item}">
+            <ProMarker :isFree="!item.pro" :video="item" />
+          </template>
+          <template #item.actions="{item}">
+            <td @click.stop class="non-clickable">
+              <v-btn small :to="`/watch/${item.id}`">Watch</v-btn>
+              <v-btn small :to="`/admin/videos/${item.id}/edit`">Edit</v-btn>
+              <v-btn small @click="deleteVideo(item)">Delete</v-btn>
+            </td>
+          </template>
+          <template #expanded-item="{headers,item}">
+            <td :colspan="headers.length">
+              <v-row>
+                <v-col cols="12" md="4">
+                  <VideoWatch :video="item" />
+                </v-col>
+                <v-col cols="12" md="8">
+                  <h1>{{item.name}}</h1>
+                  <MarkdownDisplay :markdown="item.description" />
+                </v-col>
+              </v-row>
+            </td>
+          </template>
+        </v-data-table>
+        <template v-if="isSecondTableAvailable">
+          <div class="headline">
+            {{ `Videos with '${search}' in the code summary or transcription` }}
+          </div>
+          <v-data-table :items="itemsFilteredBySummary"
```

Comment:
This is pretty much the same as the first table except for some variables we're feeding in... let's put it in a component.